src/reflect/scala/reflect/internal/Symbols.scala

```diff
@@ -2399,11 +2399,12 @@ trait Symbols extends api.Symbols { self: SymbolTable =>
      *
      *  @param baseClass is a base class of this symbol's owner.
      */
-    final def overriddenSymbol(baseClass: Symbol): Symbol = (
+    final def overriddenSymbol(baseClass: Symbol): Symbol = {
       // concrete always overrides abstract, so don't let an abstract definition
       // claim to be overriding an inherited concrete one.
-      matchingInheritedSymbolIn(baseClass) filter (res => res.isDeferred || !this.isDeferred)
-    )
+      val matching = matchingInheritedSymbolIn(baseClass)
+      if (isDeferred) matching.filter(res => res.isDeferred) else matching
```

Comment:
Aha. However, a disturbing lack of underscore.

Maybe clearer as a match or test. `if (matching != NoSymbol && (!isDeferred || matching.isDeferred))` etc.

Maybe it's time for `-Xlint:capture`.

Reply:
I'm guessing that `matching` could be an `OverloadedSymbol` in erroneous code. We shouldn't call `isDeferred` on that directly.